src/context/AuthContext.tsx

```diff
@@ -0,0 +1,103 @@
+import { IGroup } from '@/types/group';
+import { IUser } from '@/types/user';
+import { getGroupProfile } from '@/utils/group';
+import { getUserProfile } from '@/utils/user';
+import { useRouter } from 'next/router';
+
+import {
+    ReactNode,
+    createContext,
+    useCallback,
+    useContext,
+    useEffect,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+
+export interface IAuthContext {
+    isReady: boolean;
+    isAuthenticated: boolean;
+    user?: IUser;
+    group?: IGroup;
+    login: () => void;
+    logout: () => void;
+    refreshContext: () => Promise<void>;
+}
+
+export const AuthContext = createContext({} as IAuthContext);
+
+export const useAuth = () => useContext(AuthContext);
+
+const AuthProvider = ({ children }: { children: ReactNode }) => {
+    const [isReady, setIsReady] = useState(false);
+    const [isAuthenticated, setIsAuthenticated] = useState(false);
+    const [user, setUser] = useState<IUser>();
+    const [group, setGroup] = useState<IGroup>();
+    const isFetching = useRef(false);
+    const router = useRouter();
+
+    const refreshContext = useCallback(async () => {
+        if (isFetching.current) {
+            return;
+        }
+        isFetching.current = true;
+
+        const token = localStorage.getItem('token');
+        if (token) {
+            const userProfile = await getUserProfile();
+            if (!userProfile) {
+                localStorage.clear();
+                window.location.href = '/login';
+                return;
+            }
+
+            const groupProfile = await getGroupProfile(router.locale);
+            if (groupProfile) {
+                setGroup(groupProfile);
+            }
+
+            setUser(userProfile);
+            setIsAuthenticated(true);
+        }
+
+        isFetching.current = false;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
```

Comment:
Could you check if it's better to leave lines 65 - 66 as they are or add the missing dependency `router.locale` in the square brackets? Like this: line 66: `}, [router.locale]);` ?

I'm not sure if adding the dependency will cause any bugs, so it would be great if you could check for us!